portafly/src/auth/AuthContext.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { createContext, useState } from 'react'
+
+interface IAuthContext {
+  authToken: string | null,
+  setAuthToken: (l: string) => void
+}
+
+const AuthContext = createContext<IAuthContext>({
+  authToken: '',
+  setAuthToken: () => {}
+})
+
+const AuthProvider: React.FunctionComponent = ({ children }) => {
+  const existingToken = localStorage.getItem('token')
+  const [authToken, setAuthToken] = useState(existingToken && JSON.parse(existingToken))
+
+  const setToken = (token: string) => {
+    localStorage.setItem('token', JSON.stringify(token))
+    setAuthToken(token)
+  }
+  return (
+    <AuthContext.Provider value={{ authToken, setAuthToken: setToken }}>
+      {children}
+    </AuthContext.Provider>
+  )
+}
+
+export { AuthProvider, AuthContext }
```

Comment:
I'm not sure about this but I think the Context itself should not be exported, if we're going to have a hook. Instead, to promote encapsulation this what I would do:
```js
// AuthContext.tsx
const AuthContext = createContext(...)
const AuthProvider = () => ...
const useAuth = () => ..

export { AuthProvider, useAuth }
```
We end up with a single file to export from and we restraint AuthContext to a single instance (ContextProvider). This is what I'm doing for the `AlertsContext` ([link](https://github.com/3scale/porta/blob/2d65f4f0357ef025d3617d715c189f996d5cef9f/portafly/src/components/util/AlertsContext.tsx#L1)).

Reply:
Could be in the same file as it is now, yeah.